### PR TITLE
change commands and helpers

### PR DESCRIPTION
**Pull request by mimmi20**
I added some missing type hints and a return value for the commands. I also change the parameter for the run name to be the same in all commands.

### CODE REVIEW

File: src/Command/Analyze.php [hunk 1]
```diff
@@ -43,7 +43,7 @@ protected function configure(): void
             ->setHelp('');
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output): void
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (jaydiablo) [inline src/Command/Analyze.php]: Why nullable? The only returns I see now are ints.

File: src/Command/Benchmark.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ protected function configure(): void
             ->setHelp('Runs the selected parsers against a list of useragents (provided in the passed in "file" argument). By default performs just one iteration per parser but this can be configured with the "--iterations" option.  Reports the time taken and memory use of each parser.');
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output): void
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (jaydiablo) [inline src/Command/Benchmark.php]: Same for this one.

File: src/Command/Compare.php [hunk 3]
```diff
@@ -20,7 +20,7 @@ protected function configure(): void
             ->setHelp('This command is a "meta" command that will execute the Test, Normalize and Analyze commands in order');
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output): void
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (jaydiablo) [inline src/Command/Compare.php]: Will `$command->run` ever return `null`?

File: src/Command/Normalize.php [hunk 4]
```diff
@@ -23,7 +23,7 @@ protected function configure(): void
             ->setHelp('');
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output): void
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (jaydiablo) [inline src/Command/Normalize.php]: nullable?

File: src/Command/Parse.php [hunk 5]
```diff
@@ -24,20 +24,20 @@ protected function configure(): void
...
-            ->addOption('name', null, InputOption::VALUE_OPTIONAL, 'Name of the run, for storing results')
             ->addOption('csv', null, InputOption::VALUE_NONE, 'Outputs CSV without showing CLI table')
-            ->addOption('csv-file', null, InputOption::VALUE_OPTIONAL, 'File name to output CSV data to')
             ->addOption('no-output', null, InputOption::VALUE_NONE, 'Disables output after parsing, useful when chaining commands')
+            ->addOption('csv-file', null, InputOption::VALUE_OPTIONAL, 'File name to output CSV data to, implies the options "csv" and "no-output"')
             ->setHelp('Parses the useragent strings (one per line) from the passed in file and outputs the parsed properties.');
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output): void
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (jaydiablo) [inline src/Command/Parse.php]: nullable?

File: src/Command/Test.php [hunk 6]
```diff
@@ -29,11 +29,11 @@ protected function configure(): void
     {
         $this->setName('test')
             ->setDescription('Runs test against the parsers')
-            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the test run, if omitted will be generated from date')
+            ->addArgument('run', InputArgument::OPTIONAL, 'The name of the test run, if omitted will be generated from date')
             ->setHelp('Runs various test suites against the parsers to help determine which is the most "correct".');
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output): void
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (jaydiablo) [inline src/Command/Test.php]: nullable?

File: src/Command/Analyze.php (hunk 1 above)
Comment (mimmi20) [reply src/Command/Analyze.php]: The base function defines `@return null|int null or 0 if everything went fine, or an error code`. I wanted to respect this

Comment (mimmi20) [reply src/Command/Analyze.php]: fixed

File: src/Command/Benchmark.php (hunk 2 above)
Comment (mimmi20) [reply src/Command/Benchmark.php]: fixed

File: src/Command/Compare.php (hunk 3 above)
Comment (mimmi20) [reply src/Command/Compare.php]: fixed

File: src/Command/Normalize.php (hunk 4 above)
Comment (mimmi20) [reply src/Command/Normalize.php]: fixed

File: src/Command/Parse.php (hunk 5 above)
Comment (mimmi20) [reply src/Command/Parse.php]: fixed

File: src/Command/Test.php (hunk 6 above)
Comment (mimmi20) [reply src/Command/Test.php]: fixed

**Review by jaydiablo — APPROVED** [review]

**Comment by jaydiablo** [comment]
Thanks!

**Pull request merged**